x/concentrated-liquidity/pool_test.go

```diff
@@ -11,47 +11,51 @@ import (
 
 func (s *KeeperTestSuite) TestCalcOutAmtGivenIn() {
 	ctx := s.Ctx
-	pool, err := s.App.ConcentratedLiquidityKeeper.CreateNewConcentratedLiquidityPool(ctx, 1, "eth", "usdc", sdk.MustNewDecFromStr("70.710678"), sdk.NewInt(85176))
+
+	currPrice := sdk.NewDec(5000)
+	currSqrtPrice, err := currPrice.ApproxSqrt()
+	s.Require().NoError(err)
+	currTick := cl.PriceToTick(currPrice)
+	lowerPrice := sdk.NewDec(4545)
 	s.Require().NoError(err)
-	s.SetupPosition(pool.Id)
+	lowerTick := cl.PriceToTick(lowerPrice)
+	upperPrice := sdk.NewDec(5500)
+	s.Require().NoError(err)
+	upperTick := cl.PriceToTick(upperPrice)
 
-	// test asset a to b logic
-	tokenIn := sdk.NewCoin("eth", sdk.NewInt(133700))
-	tokenOutDenom := "usdc"
-	swapFee := sdk.NewDec(0)
-	minPrice := sdk.NewDec(4500)
-	maxPrice := sdk.NewDec(5500)
+	amount0 := sdk.NewInt(1)
+	amount1 := sdk.NewInt(5000)
 
-	amountOut, _, _, _, err := s.App.ConcentratedLiquidityKeeper.CalcOutAmtGivenIn(ctx, tokenIn, tokenOutDenom, swapFee, minPrice, maxPrice, pool.Id)
+	// create pool
+	pool, err := s.App.ConcentratedLiquidityKeeper.CreateNewConcentratedLiquidityPool(ctx, 1, "eth", "usdc", currSqrtPrice, currTick)
 	s.Require().NoError(err)
-	s.Require().Equal(sdk.NewDec(666975610).String(), amountOut.Amount.ToDec().String())
-
-	// test asset b to a logic
-	tokenIn = sdk.NewCoin("usdc", sdk.NewInt(4199999999))
-	tokenOutDenom = "eth"
-	swapFee = sdk.NewDec(0)
 
-	amountOut, _, _, _, err = s.App.ConcentratedLiquidityKeeper.CalcOutAmtGivenIn(ctx, tokenIn, tokenOutDenom, swapFee, minPrice, maxPrice, pool.Id)
+	// add position
+	_, _, _, err = s.App.ConcentratedLiquidityKeeper.CreatePosition(s.Ctx, pool.Id, s.TestAccs[0], amount0, amount1, sdk.ZeroInt(), sdk.ZeroInt(), lowerTick.Int64(), upperTick.Int64())
 	s.Require().NoError(err)
-	s.Require().Equal(sdk.NewDec(805287), amountOut.Amount.ToDec())
 
-	// test asset b to a logic
-	tokenIn = sdk.NewCoin("usdc", sdk.NewInt(42000000))
-	tokenOutDenom = "eth"
-	swapFee = sdk.NewDec(0)
+	// swapping parameters used for test
+	tokenIn := sdk.NewCoin("usdc", sdk.NewInt(42))
+	tokenOutDenom := "eth"
+	swapFee := sdk.ZeroDec()
+	priceLimit := sdk.NewDec(5004)
+	tokenIn, tokenOut, updatedTick, updatedLiquidity, err := s.App.ConcentratedLiquidityKeeper.CalcOutAmtGivenIn(s.Ctx, tokenIn, tokenOutDenom, swapFee, priceLimit, pool.Id)
+	expectedTokenIn := sdk.NewCoin("usdc", sdk.NewInt(0))
+	expectedTokenOut := sdk.NewCoin("eth", sdk.NewInt(42))
```

Comment:
Shouldn't both of these just be returning as 10^6 which would solve the truncation issue?

Reply:
Wdym?

Reply:
No, what I am saying is all sdk.Coin types use 10^6 as the amount. Also having 0 usdc in and 42 eth out doesn't make sense right? I could be misunderstanding

Reply:
For the second statement, I added it to the list of the chores https://github.com/osmosis-labs/osmosis/issues/3178

Reply:
> No, what I am saying is all sdk.Coin types use 10^6 as the amount

Hrrm srry Im still not fully understanding, are you proposing the numbers here should be 4200000?

Reply:
oh its the delta, okay that changed my position. With that being said, this implies the tokenOut delta is .000042 eth right?

Reply:
Im actually getting 0.008396714104746079